enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def my_each
+    each do |value|
+      yield(value)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |value|
+      yield(value, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    each do |value|
+      new_array.push(value) if yield(value)
+    end
+    new_array
+  end
+
+  def my_all?
+    flag = true
+    each do |value|
+      flag = yield(value)
+      break unless flag
+    end
+    flag
+  end
+
+  def my_any?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    flag
+  end
+
+  def my_none?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    !flag
+  end
```

Comment:
The issues with **my_none?** and **my_any?** are similar to the ones in **my_all?**. the methods should be able to
 * [ ] Handle the cases where no block is given (Hint: `block_given?`)
 * [ ]  Accept primitives, classes (i.e Integer, String, etc...) and  Regex Expressions as arguments.